core/src/main/kotlin/com/bumble/appyx/core/children/ChildNodeCreationManager.kt

```diff
@@ -0,0 +1,178 @@
+package com.bumble.appyx.core.children
+
+import androidx.lifecycle.coroutineScope
+import com.bumble.appyx.core.modality.AncestryInfo
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.navigation.RoutingKey
+import com.bumble.appyx.core.node.ParentNode
+import com.bumble.appyx.core.node.build
+import com.bumble.appyx.core.state.MutableSavedStateMap
+import com.bumble.appyx.core.state.SavedStateMap
+import com.bumble.appyx.utils.customisations.NodeCustomisationDirectory
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.flow.updateAndGet
+import kotlinx.coroutines.launch
+
+internal class ChildNodeCreationManager<Routing : Any>(
+    private var savedStateMap: SavedStateMap?,
+    private val customisations: NodeCustomisationDirectory,
+    private val childMode: ChildEntry.ChildMode,
+    private val keepMode: ChildEntry.KeepMode,
+) {
+    private val _children =
+        MutableStateFlow<Map<RoutingKey<Routing>, ChildEntry<Routing>>>(emptyMap())
+    val children: StateFlow<ChildEntryMap<Routing>> = _children.asStateFlow()
+    private lateinit var parentNode: ParentNode<Routing>
+
+    fun launch(parentNode: ParentNode<Routing>) {
+        this.parentNode = parentNode
+        savedStateMap.restoreChildren()?.also { restoredMap ->
+            _children.update { restoredMap }
+            savedStateMap = null
+        }
+        parentNode.lifecycle.coroutineScope.launch {
+            // TODO Extract as strategy? Then easy to miss bug fixes to both.
+            parentNode.navModel.screenState.collect { state ->
+                // TODO Do not calculate it if wrong keep mode
+                val navModelOnScreenKeys = state.onScreen
+                    .mapTo(HashSet(state.onScreen.size, 1f)) { element -> element.key }
+                val navModelOffScreenKeys = state.offScreen
+                    .mapTo(HashSet(state.offScreen.size, 1f)) { element -> element.key }
+                val navModelKeys = navModelOnScreenKeys + navModelOffScreenKeys
```

Comment:
I imagined it in a different way. Similarly to `onScreen/offScreen`, can't we expose `elementsToSave/elementsToRestore` and implement a strategy in `BaseNavModel` which will do the mapping of the elements with default strategy of mapping `onScreen/offScreen` to `elementsToRestore/elementsToSave`?

Reply:
I can also imagine that any particular Node can also have a flag to reflect the information if it needs to stay in the memory or not